fds/services/git_service.py

```diff
@@ -57,3 +58,9 @@ def push(remote: str, ref: str) -> Any:
                     raise Exception("No git branch found to push to")
                 push_cmd.append(curr_branch)
         execute_command(push_cmd, capture_output=False)
+
+    def clone(self, url: str, folder_name: Optional[str]) -> Any:
+        if folder_name is None or folder_name == "":
+            folder_name = get_git_repo_name_from_url(url)
```

Comment:
This actually seems unnecessary - `git clone` is smart enough to handle this on its own.
Try it yourself.

Reply:
yes i over thought about this, i'll let git handle this

Reply:
Still not fixed I think.
You can just run `git clone <url>` without passing any `folder_name` in this case.